### ISSUE TITLE
midi qol advantage on concentration flag doesn't seem to roll with advantage

### ISSUE THREAD

**Issue by tposney** [tposney/midi-qol#536]
In GitLab by @danielmoffatt on Sep 21, 2021, 19:24

Hi, I'm having an issue using flags.midi-qol.advantage.concentration (ADD 1) via a DAE effect. When I cause the concentrating target to make a concentration check, whilst the flavor text in the chat window does say advantage only one die roll is made.

**Comment by tposney** [comment]
In GitLab by @endlessgrit on Sep 21, 2021, 23:59

Try custom 1, not add 1. Works for me

**Comment by tposney** [comment]
In GitLab by @danielmoffatt on Sep 22, 2021, 24:14

This is with custom 1 also. Looks like advantage but only indicates one roll...not convinced it is doing it as I've had some real low results.

![ConcCheck](https://my-gitlab-bucket.s3.amazonaws.com/640711674/c8e737f919de95436296939b106eb34b4b94c8620212f8e78ffcf3a388dd36a8/ConcCheck.jpg)

**Comment by tposney** [comment]
In GitLab by @endlessgrit on Sep 22, 2021, 24:18

![warcsas](https://my-gitlab-bucket.s3.amazonaws.com/640711674/6d44cbaccf631b9d530558c0a3e250f43d016bb1969433839186cc87599de0da/warcsas.PNG)

**Comment by tposney** [comment]
In GitLab by @endlessgrit on Sep 22, 2021, 24:18

I put the effect as a passive buff on the character

**Comment by tposney** [comment]
In GitLab by @danielmoffatt on Sep 22, 2021, 24:40

Cheers for the response. Very odd though that's exactly what I have here.

![warcaster](https://my-gitlab-bucket.s3.amazonaws.com/640711674/0f5083a660ac6bed863daf8727b4e11e0c3893f5c80a49c2c97037076744bad3/warcaster.jpg)

**Comment by tposney** [comment]
Can you export the actor for me? I tried with a passive effect from an item and it worked fine. There must be something subtle I'm missing. Easiest if I test on the actor you are using. Also, can you export your midi-qol settings and attach them, to make sure I'm not missing something.

**Comment by tposney** [comment]
In GitLab by @danielmoffatt on Sep 23, 2021, 02:28

Hi Tim, here it is. I've tried override, custom and add 1 and all given the same one dice only result.

[fvtt-Actor-hamon-kost.json](https://my-gitlab-bucket.s3.amazonaws.com/640711674/2e600dda5cbad1a64fb60b6dd006c8f628b49318168689e6b118e5c901ec6c6a/fvtt-Actor-hamon-kost.json)

**Comment by tposney** [comment]
How is the saving throw being triggered? I.e. how is the saving throw being rolled, is the player just clicking consitution save on their character sheet, or is the roll auto rolled by midi or something like LMRTFY?

In my test world this is working with the actor you sent.

**Comment by tposney** [comment]
In GitLab by @danielmoffatt on Sep 24, 2021, 24:15

Hi, thanks for looking into this. Attached are my midi setting. It's an npc linked actor set to use LMRTFY. I'm running on forge with the latest module versions (Midi-QOL v0.8.59, DAE v0.8.54, LMRTFY v1.19).

![MidiSettings](https://my-gitlab-bucket.s3.amazonaws.com/640711674/69a65d58baad591c3df1b38fee83753bc51a0b7d8523393fb6db77993dbf51eb/MidiSettings.jpg)

**Comment by tposney** [comment]
Finally! I know what is going on. And it brings up the fact that the documentation for the midi settings is not as good as it could be and that LMRTFY has changed since I last looked,

First the intention: LMRTFY can be called 2 ways, LMRTFY the roll is done as specified by midi-qol (including adv/dis) and LMRTFY + Query where the player is able to choose adv/dis.

The way LMRTFY is implemented I can either set adv/dis (LMRTFY) or let you choose (LMRTFY + Query), but not both.

So for the advantage to be set you need to use LMRTFY rather than LMRTFY + Query.

For the LMRTFY + QUERY case, the player will be prompted for adv/dis or (if fastforward abiltiy rolls is set they won't be prompted but the player can use alt/ctl for advantage/disadvantage).

**Comment by tposney** [comment]
In GitLab by @danielmoffatt on Sep 24, 2021, 19:16

Hi Tim, I just did some testing based on your response above. Even with the setting of LMRTY I get the same single dice roll (I had wondered what the +query did though lol). I also am using Monks Token Bar so I switched to that and as shown again it is a single roll only strangely.

![UsingMonks](https://my-gitlab-bucket.s3.amazonaws.com/640711674/788ad9267d26642eb9fae4c846e2a6f43303dddcede7511f2be87cfe07a59894/UsingMonks.jpg)

**Comment by tposney** [comment]
In GitLab by @danielmoffatt on Sep 24, 2021, 19:21

Also the auto setting seems to only roll 1 dice from what I can see

![UsingAuto](https://my-gitlab-bucket.s3.amazonaws.com/640711674/7af9192d1ef695d3b2304dc833c77f7a36201a22d9a722aad81361f483319ebb/UsingAuto.jpg)

**Comment by tposney** [comment]
Well that took some digging. Before going into the details I thought I should explain the history of this whole saving throw piece. Originally I wrote it so that midi would always roll saving throws automatically for everyone. Then there was a request for players to be able to roll their own saves (and hence the player roll save options), then monk's token bar came around and I added that from some requests.

Later people wanted to be able to do the rolls for the GM by hand. Since LMRTFY does not let you change adv/disadv I set the GM rolls to always be LMRTFY + Query, even though the selection is not that - I have no idea why I chose to do that, rather than setting the options correctly.

So, long story short, rolls by a non gm-player (which was the original intent) do roll as I described before. Rolls by the GM, which includes player characters if they are not connected to the game use LMRTFY+Query.

I'll put in a change in 0.8.62 to give the GM the fulll range of options.

Monk's token bar does not support setting advantage for the save, which really should have been in the Readme/Changelog. I sort of assumed that people who used Monk's token bar would know that advantage could not be set by the GM.

For the second case (the auto roll), the top line of the tooltip is the roll that was done so 2d20kh+1 (which is advantage) and the second line is the result of the roll i.e of the two dice the highest was a 6 and 1 was added.

So, when 0.8.62 comes out it should do what you are expecting.
